### DISCUSSION TITLE
Issue with Updating Auth User Profile Information

### DISCUSSION THREAD

**Discussion by ChoaibMouhrach** [nextauthjs/next-auth#9979] [Help]
Issue with Updating Auth User Profile Information

Hi everyone,

I'm currently working on a profile page where I want to update the authenticated user's profile information, specifically the email and name. I'm using a magic link provider (Email provider) for authentication.

The flow involves sending two confirmation emails to the user's inbox. Once the links in these emails are clicked, the users are redirected to a route handler that updates their profile. After the update, I redirect them to the profile page. However, I've noticed that the profile still displays the old email.

I believe the issue lies in updating the session to reflect the newly added data. I've attempted to use something like `const { update } = useSession()` on the backend, similar to the update function. In my route handler, I tried the following code:

```typescript
await fetch(getUrl("/api/auth/session", request.url), {
    method: "POST",
    headers: {
      "content-type": "application/json",
    },
    body: JSON.stringify({
      csrfToken: await getCsrfToken(),
      data: {
        email: emailSwitcher.email,
      },
    }),
});
```

However, this results in a `400 [object Object]` error. I suspect there might be an issue with the data format or some missing information.

Any guidance on how to properly update the session or insights into resolving the 400 error would be greatly appreciated!

Thank you in advance for your help.

**Comment by Bernardomennndes** [comment]
Are you using this POST fetch on a route handler? The endpoint '/api/auth/session' is suposed to be used on the client side of your application.
